baremetal/metal3machine_manager.go

```diff
@@ -149,6 +151,43 @@ func (m *MachineManager) role() string {
 	return bmRoleNode
 }
 
+// SetPauseAnnotation sets the pause annotations on associated bmh
+func (m *MachineManager) SetPauseAnnotation(ctx context.Context) error {
+	// look for associated BMH
+	host, err := m.getHost(ctx)
+	if err != nil {
+		m.setError("Failed to get a BaremetalHost for the Metal3Machine",
+			capierrors.CreateMachineError,
+		)
+		return err
+	}
+	if host == nil {
+		m.Log.Info("BaremetalHost not associated")
+		return nil
+	}
+
+	annotations := host.GetAnnotations()
+
+	if annotations != nil {
+		if _, ok := annotations[bmh.PausedAnnotation]; ok {
+			if host.Annotations[bmh.PausedAnnotation] != pausedAnnotationKey {
+				m.Log.Info("BaremetalHost is already paused by user")
```

Comment:
If you don't set it yourself, what if the user removes its pause annotation ? What will be the flow ?

Reply:
A change in the bmh will reconcile CAPM3 machine and then  if the cluster/CAPM3 machine is still paused, the annotation will be added by CAPM3

Reply:
yes, but in between the BMH will have been reconciled, at least once. That could be very problematic

Reply:
Hmmm, if it has to be done through CAPM3, then what else can we do?

Reply:
Maybe modify the annotation to add the CAPM3 part to the user input ? so that when the cluster is not paused anymore, you are still able to restore the pause field of the user.

Reply:
Please check the removal logic and comment if its ok